lib/app_profiler.rb

```diff
@@ -60,17 +63,75 @@ module Viewer
   mattr_reader :after_process_queue, default: nil
 
   class << self
-    def run(*args, &block)
-      Profiler.run(*args, &block)
+    def run(*args, with_backend: nil, **kwargs, &block)
+      yield unless acquire_run_lock
```

Comment:
Running stackprof and vernier concurrently can cause  a segmentation fault, which is why this mutex is used to be certain that we cannot have a race condition leading to both profilers running at the same time. Previously, we relied on stackprof itself to prevent multiple concurrent sessions.